x-pack/plugins/fleet/public/applications/integrations/sections/epm/components/package_list_grid.tsx

```diff
@@ -113,116 +148,199 @@ export const PackageListGrid: FunctionComponent<Props> = ({
-    gridContent = <Loading />;
-  } else {
-    gridContent = (
-      <GridColumn
-        list={filteredPromotedList}
-        showMissingIntegrationMessage={showMissingIntegrationMessage}
-        showCardLabels={showCardLabels}
-      />
-    );
-  }
+  const splitSubcategories = (
+    subcategories: CategoryFacet[] | undefined
+  ): { visibleSubCategories?: CategoryFacet[]; hiddenSubCategories?: CategoryFacet[] } => {
+    if (!subcategories) return {};
+    else if (subcategories && subcategories?.length < MAX_SUBCATEGORIES_NUMBER) {
+      return { visibleSubCategories: subcategories, hiddenSubCategories: [] };
+    } else if (subcategories && subcategories?.length >= MAX_SUBCATEGORIES_NUMBER) {
+      return {
+        visibleSubCategories: subcategories.slice(0, MAX_SUBCATEGORIES_NUMBER),
+        hiddenSubCategories: subcategories.slice(MAX_SUBCATEGORIES_NUMBER),
+      };
+    }
+    return {};
+  };
 
-  return (
-    <>
-      <div ref={menuRef}>
-        <EuiFlexGroup
-          alignItems="flexStart"
-          gutterSize="xl"
-          data-test-subj="epmList.integrationCards"
+  const splitSubcat = splitSubcategories(availableSubCategories);
+  const { visibleSubCategories } = splitSubcat;
+  const hiddenSubCategoriesItems = useMemo(() => {
+    return splitSubcat?.hiddenSubCategories?.map((subCategory) => {
+      return (
+        <EuiContextMenuItem
+          key={subCategory.id}
+          onClick={() => {
+            onSubCategoryClick(subCategory.id);
+            closePopover();
+          }}
         >
-          <EuiFlexItem grow={1} className={isSticky ? 'kbnStickyMenu' : ''}>
-            {controlsContent}
-          </EuiFlexItem>
-          <EuiFlexItem grow={5}>
-            <EuiFieldSearch
-              data-test-subj="epmList.searchBar"
-              placeholder={i18n.translate('xpack.fleet.epmList.searchPackagesPlaceholder', {
-                defaultMessage: 'Search for integrations',
-              })}
-              value={searchTerm}
-              onChange={(e) => onQueryChange(e)}
-              isClearable={true}
-              incremental={true}
-              fullWidth={true}
-              prepend={
-                selectedCategoryTitle ? (
-                  <EuiText
-                    data-test-subj="epmList.categoryBadge"
-                    size="xs"
+          {subCategory.title}
+        </EuiContextMenuItem>
+      );
+    });
+  }, [onSubCategoryClick, splitSubcat.hiddenSubCategories]);
+
+  return (
+    <EuiFlexGroup alignItems="flexStart" gutterSize="xl" data-test-subj="epmList.integrationCards">
+      <EuiFlexItem
+        data-test-subj="epmList.controlsSideColumn"
+        grow={1}
+        style={{
+          position: 'sticky',
+          top: 0,
+          zIndex: 100,
+        }}
+      >
+        <ControlsColumn controls={controls} title={title} />
+      </EuiFlexItem>
+      <EuiFlexItem grow={5} data-test-subj="epmList.mainColumn">
```

Comment:
nit: Should we split some of these UI elements into smaller components and import them? This file is getting quite big

Reply:
It would be good, yes. I stopped there to avoid doing an even bigger PR.